### ISSUE TITLE
[Bug]: [rolling upgrade] Enable mmap for rolling upgrades from 2.3 to 2.4 versions, upgrade balance channel stuck

### ISSUE THREAD

**Issue by elstic** [milvus-io/milvus#31964]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3.12 -> 2.4
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Version 2.3.12 inserted 25m 128d of data, loaded the rolling upgrade, and the rolling upgrade took more than 25 minutes, and only balanced a segment for 7 minutes. 

client argo task : fouramf-wfblr
upgrade argo task: fouramf-gzsv9

![image](https://github.com/milvus-io/milvus/assets/48523564/2ae30994-9e8d-4e34-ade5-37a355044a7a)

Rolling upgrades take approximately 25 minutes: 
2024-04-07 08:51:30~ 09:16:00 
 
before upgrade:
```
fouramf-wfblr-op-89-8641-milvus-datacoord-7954bfbcbf-678m7       Running     0            50m     10.104.19.52      4am-node28     
fouramf-wfblr-op-89-8641-milvus-datanode-55454dbb9b-qlnwv        Running     0            50m     10.104.29.197     4am-node35     
fouramf-wfblr-op-89-8641-milvus-indexcoord-5f8db986bc-945c6      Running     0            50m     10.104.29.196     4am-node35     
fouramf-wfblr-op-89-8641-milvus-indexnode-76c498f67b-mndff       Running     0            50m     10.104.31.247     4am-node34     
fouramf-wfblr-op-89-8641-milvus-proxy-cf9bb84fb-zls5x            Running     0            50m     10.104.29.193     4am-node35     
fouramf-wfblr-op-89-8641-milvus-querycoord-56454bb9cf-pr4b8      Running     0            50m     10.104.29.194     4am-node35     
fouramf-wfblr-op-89-8641-milvus-querynode-0-669cbf7c8c-6dcdt     Running     0            50m     10.104.19.53      4am-node28     
fouramf-wfblr-op-89-8641-milvus-rootcoord-76ccc75dd5-82wz2       Running     0            50m     10.104.29.195     4am-node35     
fouramf-wfblr-op-89-8641-etcd-0                                  Running     0            54m     10.104.32.236     4am-node39     
fouramf-wfblr-op-89-8641-etcd-1                                  Running     0            54m     10.104.34.216     4am-node37     
fouramf-wfblr-op-89-8641-etcd-2                                  Running     0            54m     10.104.27.250     4am-node31     
fouramf-wfblr-op-89-8641-pulsar-bookie-0                         Running     0            54m     10.104.34.219     4am-node37     
fouramf-wfblr-op-89-8641-pulsar-bookie-1                         Running     0            54m     10.104.32.238     4am-node39     
fouramf-wfblr-op-89-8641-pulsar-bookie-2                         Running     0            54m     10.104.27.254     4am-node31     
fouramf-wfblr-op-89-8641-pulsar-broker-0                         Running     0            54m     10.104.32.233     4am-node39     
fouramf-wfblr-op-89-8641-pulsar-proxy-0                          Running     0            54m     10.104.34.211     4am-node37     
fouramf-wfblr-op-89-8641-pulsar-recovery-0                       Running     0            54m     10.104.1.190      4am-node10     
fouramf-wfblr-op-89-8641-pulsar-zookeeper-0                      Running     0            54m     10.104.34.218     4am-node37     
fouramf-wfblr-op-89-8641-pulsar-zookeeper-1                      Running     0            53m     10.104.32.240     4am-node39     
fouramf-wfblr-op-89-8641-pulsar-zookeeper-2                      Running     0            53m     10.104.31.239     4am-node34     
fouramf-wfblr-op-89-8641-minio-0                                 Running     0            54m     10.104.32.237     4am-node39     
fouramf-wfblr-op-89-8641-minio-1                                 Running     0            54m     10.104.34.217     4am-node37     
fouramf-wfblr-op-89-8641-minio-2                                 Running     0            54m     10.104.27.251     4am-node31     
fouramf-wfblr-op-89-8641-minio-3                                 Running     0            54m     10.104.28.9       4am-node33
```

after upgrade:
```
fouramf-wfblr-op-89-8641-etcd-0                                   1/1     Running                           0                 109m    10.104.32.236   4am-node39   <none>           <none>
fouramf-wfblr-op-89-8641-etcd-1                                   1/1     Running                           0                 109m    10.104.34.216   4am-node37   <none>           <none>
fouramf-wfblr-op-89-8641-etcd-2                                   1/1     Running                           0                 109m    10.104.27.250   4am-node31   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-datacoord-5967b4bb85-mbszh        1/1     Running                           0                 55m     10.104.1.242    4am-node10   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-datanode-69845c756-dfs69          1/1     Running                           0                 55m     10.104.13.101   4am-node16   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-indexcoord-f4c74b7c5-xbbxg        1/1     Running                           0                 55m     10.104.1.245    4am-node10   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-indexnode-6f96cd9d7b-5wjd2        1/1     Running                           0                 55m     10.104.1.244    4am-node10   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-proxy-7b5d79dbcb-wcmgw            1/1     Running                           0                 55m     10.104.1.241    4am-node10   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-querycoord-6488d65b87-ngvks       1/1     Running                           0                 55m     10.104.1.243    4am-node10   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-querynode-1-58cc89fb78-fqj8f      1/1     Running                           0                 54m     10.104.20.109   4am-node22   <none>           <none>
fouramf-wfblr-op-89-8641-milvus-rootcoord-56c6db659c-hbpb2        1/1     Running                           0                 55m     10.104.1.240    4am-node10   <none>           <none>
fouramf-wfblr-op-89-8641-minio-0                                  1/1     Running                           0                 109m    10.104.32.237   4am-node39   <none>           <none>
fouramf-wfblr-op-89-8641-minio-1                                  1/1     Running                           0                 109m    10.104.34.217   4am-node37   <none>           <none>
fouramf-wfblr-op-89-8641-minio-2                                  1/1     Running                           0                 109m    10.104.27.251   4am-node31   <none>           <none>
fouramf-wfblr-op-89-8641-minio-3                                  1/1     Running                           0                 109m    10.104.28.9     4am-node33   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-bookie-0                          1/1     Running                           0                 109m    10.104.34.219   4am-node37   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-bookie-1                          1/1     Running                           0                 109m    10.104.32.238   4am-node39   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-bookie-2                          1/1     Running                           0                 109m    10.104.27.254   4am-node31   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-bookie-init-c4gqk                 0/1     Completed                         0                 109m    10.104.32.231   4am-node39   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-broker-0                          1/1     Running                           0                 109m    10.104.32.233   4am-node39   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-proxy-0                           1/1     Running                           0                 109m    10.104.34.211   4am-node37   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-pulsar-init-gm5ps                 0/1     Completed                         0                 109m    10.104.32.230   4am-node39   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-recovery-0                        1/1     Running                           0                 109m    10.104.1.190    4am-node10   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-zookeeper-0                       1/1     Running                           0                 109m    10.104.34.218   4am-node37   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-zookeeper-1                       1/1     Running                           0                 108m    10.104.32.240   4am-node39   <none>           <none>
fouramf-wfblr-op-89-8641-pulsar-zookeeper-2                       1/1     Running                           0                 108m    10.104.31.239   4am-node34   <none>           <none>
```

### Expected Behavior

It's best to finish the upgrade in a few minutes

**Comment by yanliang567** [comment]
@elstic did the service interrupt during rolling upgrade? 
/assing

**Comment by elstic** [comment]
> @elstic did the service interrupt during rolling upgrade? /assing

Yes, the service was down for about 14 minutes during the upgrade:  2024-04-07 17:03:00~17:17:00

**Comment by weiliu1031** [comment]
balance channel stuck, should be fixed by #31995, please verify this with latest image

**Comment by elstic** [comment]
> balance channel stuck, should be fixed by #31995, please verify this with latest image

I think the fix is working, rolling escalation breaks from 20 minutes to 1 minute

before upgrade:
```
issue-31964-milvus-datacoord-589ccc495c-rgs6s       Running     0            1h17m     10.104.5.14       4am-node12     
issue-31964-milvus-datanode-5c8d9b47dd-c9mtt        Running     0            1h17m     10.104.5.15       4am-node12     
issue-31964-milvus-indexcoord-7d7bc57475-v8vnt      Running     0            1h17m     10.104.34.186     4am-node37     
issue-31964-milvus-indexnode-56d657b84-w6xcd        Running     0            1h17m     10.104.33.86      4am-node36     
issue-31964-milvus-proxy-9b779fccd-dpd4v            Running     0            1h17m     10.104.15.84      4am-node20     
issue-31964-milvus-querycoord-58db5c8bb7-z8sgz      Running     0            1h17m     10.104.15.83      4am-node20     
issue-31964-milvus-querynode-0-648744c95f-tmtxk     Running     0            1h17m     10.104.20.32      4am-node22     
issue-31964-milvus-rootcoord-5646dd75cc-gr872       Running     0            1h17m     10.104.23.113     4am-node27     
issue-31964-etcd-0                                  Running     0            1h21m     10.104.15.71      4am-node20     
issue-31964-etcd-1                                  Running     0            1h21m     10.104.16.89      4am-node21     
issue-31964-etcd-2                                  Running     0            1h21m     10.104.34.160     4am-node37     
issue-31964-pulsar-bookie-0                         Running     0            1h21m     10.104.15.72      4am-node20     
issue-31964-pulsar-bookie-1                         Running     0            1h21m     10.104.34.163     4am-node37     
issue-31964-pulsar-bookie-2                         Running     0            1h21m     10.104.33.83      4am-node36     
issue-31964-pulsar-broker-0                         Running     0            1h21m     10.104.16.86      4am-node21     
issue-31964-pulsar-proxy-0                          Running     0            1h21m     10.104.5.9        4am-node12     
issue-31964-pulsar-recovery-0                       Running     0            1h21m     10.104.15.65      4am-node20     
issue-31964-pulsar-zookeeper-0                      Running     0            1h21m     10.104.15.73      4am-node20     
issue-31964-pulsar-zookeeper-1                      Running     0            1h20m     10.104.23.92      4am-node27     
issue-31964-pulsar-zookeeper-2                      Running     0            1h20m     10.104.34.169     4am-node37     
issue-31964-minio-0                                 Running     0            1h21m     10.104.15.70      4am-node20     
issue-31964-minio-1                                 Running     0            1h21m     10.104.16.88      4am-node21     
issue-31964-minio-2                                 Running     0            1h21m     10.104.34.156     4am-node37     
issue-31964-minio-3                                 Running     0            1h21m     10.104.18.164     4am-node25   
```

after upgrade:
```
issue-31964-milvus-datacoord-58648cbf89-gntb6      Running     0            21m       10.104.13.225     4am-node16     
issue-31964-milvus-datanode-9fc8dbf86-wqcnz        Running     0            21m       10.104.13.224     4am-node16     
issue-31964-milvus-indexcoord-6564f696f7-l7hkk     Running     0            21m       10.104.9.240      4am-node14     
issue-31964-milvus-indexnode-65f5f6774d-q5r6s      Running     0            21m       10.104.29.218     4am-node35     
issue-31964-milvus-proxy-55b9c8987d-8hprj          Running     0            21m       10.104.4.97       4am-node11     
issue-31964-milvus-querycoord-65bf4f6f45-4ktht     Running     0            21m       10.104.4.98       4am-node11     
issue-31964-milvus-querynode-1-7c4df8d96-l4hnt     Running     0            21m       10.104.9.241      4am-node14     
issue-31964-milvus-rootcoord-6559f55849-r4jts      Running     0            21m       10.104.29.217     4am-node35     
issue-31964-etcd-0                                 Running     0            1h43m     10.104.15.71      4am-node20     
issue-31964-etcd-1                                 Running     0            1h43m     10.104.16.89      4am-node21     
issue-31964-etcd-2                                 Running     0            1h43m     10.104.34.160     4am-node37     
issue-31964-pulsar-bookie-0                        Running     0            1h43m     10.104.15.72      4am-node20     
issue-31964-pulsar-bookie-1                        Running     0            1h43m     10.104.34.163     4am-node37     
issue-31964-pulsar-bookie-2                        Running     0            1h43m     10.104.33.83      4am-node36     
issue-31964-pulsar-broker-0                        Running     0            1h43m     10.104.16.86      4am-node21     
issue-31964-pulsar-proxy-0                         Running     0            1h43m     10.104.5.9        4am-node12     
issue-31964-pulsar-recovery-0                      Running     0            1h43m     10.104.15.65      4am-node20     
issue-31964-pulsar-zookeeper-0                     Running     0            1h43m     10.104.15.73      4am-node20     
issue-31964-pulsar-zookeeper-1                     Running     0            1h42m     10.104.23.92      4am-node27     
issue-31964-pulsar-zookeeper-2                     Running     0            1h41m     10.104.34.169     4am-node37     
issue-31964-minio-0                                Running     0            1h43m     10.104.15.70      4am-node20     
issue-31964-minio-1                                Running     0            1h43m     10.104.16.88      4am-node21     
issue-31964-minio-2                                Running     0            1h43m     10.104.34.156     4am-node37     
issue-31964-minio-3                                Running     0            1h43m     10.104.18.164     4am-node25 
```
